Sourcery/Parsing/Parser.swift

```diff
@@ -297,9 +297,22 @@ final class Parser {
             })
         }
 
-        //TODO: replace typealiases of typealiases
-        //typealias Foo = Int; typealias Bar = Foo
-        //TODO: replace typealiases in inherited types
+        //! if a typealias leads to another typealias, follow through and replace with final type
+        typealiasesByNames.forEach { _, alias in
+
+            var aliasNamesToReplace = [alias.name]
```

Comment:
easier logic this way, it will reassign to the same place if there is no alias redirection, which is neglible.

What this loop does:

1. It looks whether an alias is redirecting to another typealias, and it follows through until it finds final alias, all while gathering all typealiases on the way. 
2. Replaces all intermediate typealiase names with the final typealias

Basically if you had `A -> B -> C -> D` after this loop you'd have `A -> D` `B -> D` `C -> D`